cpu_scheduling_algorithms/round_robin.cpp

```diff
@@ -0,0 +1,191 @@
+/**
+ * @file
+ * @brief Implementation of Round CPU scheduling algorithm
+ * @details
+ * Round Robin is a CPU scheduling algorithm where each process is assigned a
+ * fixed time slot in a cyclic way. It is basically the preemptive version of
+ * First come First Serve CPU Scheduling algorithm.
+ * @link https://en.wikipedia.org/wiki/Round-robin_scheduling
+ * @author [Varad Pundlik](https://github.com/varadpundlik)
+ *
+ */
+
+#include <algorithm>  // for max function
+#include <cassert>    // fot testing purpose
+#include <iostream>   //for printing results
+#include <queue>      //for job queue
+#include <string>     // for job name
+#include <utility>    //for using pair
+#include <vector>     // for tasklist of processes
+
+using std::cin;
+using std::cout;
+using std::endl;
+using std::max;
+using std::pair;
+using std::queue;
+using std::string;
+using std::vector;
+
+/** @class Job
+    @brief class to represent each process or job
+*/
+class Job {
+    int id;
+    string name;
+    int at;   // Arrival time
+    int wt{0};  // Waiting time
+    int bt;   // Burst time
+    int b1;   // Remaining burst time
+    int tat{0};  // Turnaround time
+    int ct{0};   // Completion time
```

Comment:
```suggestion
    int at;   ///< Arrival time
    int wt{0};  ///< Waiting time
    int bt;   ///< Burst time
    int b1;   ///< Remaining burst time
    int tat{0};  ///< Turnaround time
    int ct{0};   ///< Completion time
```

Please make these variable identifiers longer and detailed. currently you would need to glance at documentation just to understand what the variables stand for